### PR TITLE
Activist list paging 2

### PR DESCRIPTION
**Pull request by cbailey20043**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -129,6 +129,7 @@ func router() *mux.Router {
 	router.Handle("/event/list", alice.New(main.apiAuthMiddleware).ThenFunc(main.EventListHandler))
 	router.Handle("/event/delete", alice.New(main.apiAuthMiddleware).ThenFunc(main.EventDeleteHandler))
 	router.Handle("/activist/list", alice.New(main.apiAuthMiddleware).ThenFunc(main.ActivistListHandler))
+	router.HandleFunc("/activist/list_range", main.ActivistInfiniteScrollHandler) // temp unauthed for test
```
Comment (cbailey20043) [inline main.go]: I forgot to change this to an authed endpoint. I'll make the change before deploying...

File: frontend/ActivistList.vue [hunk 2]
```diff
@@ -460,6 +476,49 @@ export default {
...
+            this.$refs.infiniteLoading.$emit('$InfiniteLoading:complete');
+          }
+        },
+        error: () => {
+          console.warn(err.responseText);
+          flasMessage("Server error: " + err.responseText, true);
+        },
+      });
+    },
+    reset: function() {
```
Comment (samertm) [inline frontend/ActivistList.vue]: smart

File: model/activist.go [hunk 3]
```diff
@@ -88,6 +117,13 @@ type GetActivistOptions struct {
 	Hidden bool
 }
 
+//TODO Rename this
```
Comment (samertm) [inline model/activist.go]: Rename to what?

File: model/activist.go [hunk 4]
```diff
@@ -114,45 +150,64 @@ func GetActivistJSON(db *sqlx.DB, options GetActivistOptions) (ActivistJSON, err
 }
 
 func getActivistsJSON(db *sqlx.DB, options GetActivistOptions) ([]ActivistJSON, error) {
-	var activistsJSON []ActivistJSON
 	activists, err := GetActivistsExtra(db, options)
 	if err != nil {
 		return nil, err
 	}
-	for _, u := range activists {
+	activistsJSON := buildActivistJSONArray(activists)
```
Comment (samertm) [inline model/activist.go]: Nit: you can just return this directly

File: model/activist.go [hunk 5]
```diff
@@ -114,45 +150,64 @@ func GetActivistJSON(db *sqlx.DB, options GetActivistOptions) (ActivistJSON, err
...
+}
+
+func buildActivistJSONArray(activists []ActivistExtra) []ActivistJSON {
+	var activistsJSON []ActivistJSON
+
+	for _, a := range activists {
 		firstEvent := ""
-		if u.ActivistEventData.FirstEvent != nil {
-			firstEvent = u.ActivistEventData.FirstEvent.Format(EventDateLayout)
+		if a.ActivistEventData.FirstEvent != nil {
```
Comment (samertm) [inline model/activist.go]: thanks for cleaning this up haha

File: model/activist.go [hunk 6]
```diff
@@ -114,45 +150,64 @@ func GetActivistJSON(db *sqlx.DB, options GetActivistOptions) (ActivistJSON, err
 }
 
 func getActivistsJSON(db *sqlx.DB, options GetActivistOptions) ([]ActivistJSON, error) {
```
Comment (samertm) [inline model/activist.go]: Is this function used anywhere? If not, let's delete it

Comment (cbailey20043) [reply model/activist.go]: Indeed it is

**Comment by samertm** [comment]
:ship:

**Pull request merged**